packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/internal/InternalController.java

```diff
@@ -0,0 +1,47 @@
+package software.uncharted.terarium.hmiserver.controller.internal;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.annotations.tags.Tags;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import software.uncharted.terarium.hmiserver.models.data.project.Project;
+import software.uncharted.terarium.hmiserver.service.data.ProjectService;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@RequestMapping("/internal")
+@RestController
+@Slf4j
+@RequiredArgsConstructor
+@Tags(@Tag(name = "Internal", description = "Insecure internal access"))
+public class InternalController {
+	public static final String PATH = "/internal";
+
+	final ProjectService projectService;
+
+	@Operation(summary = "Gets a project by ID")
+	@ApiResponses(value = {
+		@ApiResponse(responseCode = "200", description = "Project found.", content = {
+			@Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = Project.class)) }),
+		@ApiResponse(responseCode = "500", description = "Error finding project", content = @Content),
+		@ApiResponse(responseCode = "404", description = "Project not found", content = @Content) })
+	@GetMapping("/projects/{id}")
+	public ResponseEntity<Project> getProject(
```

Comment:
For the final version, will we need to duplicate all of the endpoints we want our internal services to hit like this?

Reply:
Correct, those that are required

Reply:
I am not certain we need all that Swagger stuff... maybe it is vital for the internal calls <shrug>